core/src/main/scala/org/ensime/core/Refactoring.scala

```diff
@@ -24,7 +25,16 @@ abstract class RefactoringEnvironment(file: String, start: Int, end: Int) {
     procId: Int,
     tpe: RefactorType,
     parameters: refactoring.RefactoringParameters
-  ): Either[RefactorFailure, RefactorEffect] = {
+  )(implicit charset: Charset): Either[RefactorFailure, RefactorDiffEffect] = {
```

Comment:
Why we would need to choose different `Charset` for the diff file? And on what level? request? this class only? 

I think the `Charset` should be consistent to avoid potential bugs, in all these references in `Refactoring.scala` the `Charset` is taken from https://github.com/ensime/ensime-server/blob/master/core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala#L69